CMakeLists.txt

```diff
@@ -17,45 +19,74 @@ if("$ENV{ARCH_NAME}" STREQUAL "wormhole_b0")
     set(NORMALIZED_ARCH_NAME "wormhole")
 endif()
 
+if(DEFINED ENV{CMAKE_C_COMPILER} AND DEFINED ENV{CMAKE_CXX_COMPILER})
+    message(STATUS "Setting C and C++ compiler from environment variables")
+    set(CMAKE_C_COMPILER $ENV{CMAKE_C_COMPILER})
+    set(CMAKE_CXX_COMPILER $ENV{CMAKE_CXX_COMPILER})
+endif()
+
+if(CMAKE_CXX_COMPILER AND CMAKE_C_COMPILER)
+    message(STATUS "Using specifed C++ compiler: ${CMAKE_CXX_COMPILER}")
+    message(STATUS "Using specifed C compiler: ${CMAKE_C_COMPILER}")
+else()
+    message(STATUS "No C or C++ compiler specified, defaulting to Clang-17")
+    FIND_AND_SET_CLANG17()
+endif()
+
+list(PREPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+set(ENV{CPM_SOURCE_CACHE} "${PROJECT_SOURCE_DIR}/.cpmcache")
```

Comment:
Please respect the user's CPM cache configuration if they already have one.

Reply:
What do you mean by this? Like using the CPM cache that's already in their tt-metal repo?

Reply:
See https://github.com/tenstorrent/cpp-ttnn-project-template/blob/main/sources/sample_lib/CMakeLists.txt

Reply:
> Like using the CPM cache that's already in their tt-metal repo

No like `CPM_SOURCE_CACHE` can be set by the user from shell. The code snippet force overwrites it (ignores the user's settings). IMHO Tenstorrent should have respect the user's settings. It's their bug. Let's not replicate it.